File: Problems/Easy/155. min-stack.py
class MinStack(object):

    
    def __init__(self):
        self.stack=[]
        

    def push(self, val):
        curMin = self.getMin()
        if curMin == None or val < curMin:
            curMin = val
        self.stack.append((val, curMin));
        

    def pop(self):
        self.stack.pop()
        

    def top(self):
        if len(self.stack) == 0:
            return None
        else:
            return self.stack[-1][0]
        

    def getMin(self):
        if len(self.stack) == 0:
            return None
        else:
            return self.stack[-1][1]


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()

File: Problems/Easy/155. min-stack.cs

#NOT THE BEST SOLUTION, REFER PY FOR THE BEST SOLUTION
public class MinStack {

    public Stack<int> stack;
    public MinStack() {
        stack = new Stack<int>();
    }
    
    public void Push(int val) {
        stack.Push(val);
        
    }
    
    public void Pop() {
        stack.Pop();
    }
    
    public int Top() {
        return stack.Peek();
    }
    
    public int GetMin() {
        int min=Int32.MaxValue;
        foreach (int num in stack)
            if (num< min)
                min=num;
        return min;
    }
}
